modules/redis-cluster/main.tf

```diff
@@ -32,6 +32,14 @@ resource "google_redis_cluster" "redis_cluster" {
     }
   }
 
+  dynamic "zone_distribution_config" {
+    for_each = var.zone_distribution_config_mode != "" ? [1] : []
+    content {
+      mode = var.zone_distribution_config_mode
+      zone = var.zone_distribution_config_mode == "SINGLE_ZONE" ? var.zone_distribution_config_zone : null
+    }
+  }
+
```

Comment:
@imrannayer updated to add zone inside `zone_distribution_config` dynamic block and only add the value inside zone if the mode is `SINGLE_ZONE` as referred to here: [[+](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/redis_cluster#nested_zone_distribution_config)]

Please review again. Thanks

Reply:
@lanandra Can you plz use null instead of empty string?